charts/spire/charts/spiffe-oidc-discovery-provider/templates/hpa.yaml

```diff
@@ -1,5 +1,5 @@
 {{- if .Values.autoscaling.enabled }}
-apiVersion: autoscaling/v2beta1
+apiVersion: autoscaling/v2
```

Comment:
Can we check for the kube verison here that introduced this version, so it doesn't break for people running older clusters?

Reply:
Sounds good, will add a kube version check.

Reply:
Since this chart is for k8s 1.21 or above, I think we should at least upgrade to autoscaling/v2beta2 for people running older clusters. autoscaling/v2beta1 was deprecated in k8s 1.19 and has a different API (resource.targetAverageUtilization vs resource.target.type/resource.target.averageUtilization).

So I am thinking about checking .Capabilities.APIVersions to see if autoscaling/v2 is available and if not, go with autoscaling/v2beta2.

Reply:
Sounds good